sympy/sets/fancysets.py

```diff
@@ -236,10 +236,10 @@ def _contains(self, other):
         from sympy.solvers.solveset import solveset, linsolve
         L = self.lamda
         if self._is_multivariate():
-            solns = list(linsolve([expr - val for val, expr in zip(other, L.expr)],
-                         L.variables).args[0])
+            solns = linsolve([expr - val for val, expr in zip(other, L.expr)],
+                             L.variables).args
```

Comment:
I don't remember a use-case when self is multivariate, so not sure why tests pass or perhaps these lines are not tested well. `linsolve` returns a `FiniteSet` (when solution exist) of which the first and only argument  is a tuple:

like say:

```
In [11]: A = Matrix([[1, 2, 3], [4, 5, 6], [7, 8, 10]])

In [12]: b = Matrix([3, 6, 9])

In [13]: linsolve((A, b), [x, y, z])
Out[13]: {(-1, 2, 0)}

In [14]: solns = linsolve((A, b), [x, y, z])

In [15]: solns
Out[15]: {(-1, 2, 0)}
```
Iterating solns.args in `Line 244` will give a tuple of solution in one iteration.
```
In [23]: for soln in solns.args:
   ....:     print(soln)
   ....:     
(-1, 2, 0)
```
Where as Iterating solns.args[0] in `Line 244` will iterate over the solutions in the tuple.
```
In [24]: for soln in solns.args[0]:
   ....:     print(soln)
   ....:     
-1
2
0
```
That was my intuition for using `solns.args[0]` & of course you have to address the case for `EmptySet()` in this case.